virtio-devices/src/vhost_user/vu_common_ctrl.rs

```diff
@@ -378,4 +412,145 @@ impl VhostUserHandle {
             self.migration_support = true;
         }
     }
+
+    fn update_log_base(&mut self, last_ram_addr: u64) -> Result<Option<Arc<MmapRegion>>> {
+        // Create the memfd
+        let fd = memfd_create(
+            &ffi::CString::new("vhost_user_dirty_log").unwrap(),
+            libc::MFD_CLOEXEC | libc::MFD_ALLOW_SEALING,
+        )
+        .map_err(Error::MemfdCreate)?;
+
+        // Safe because we checked the file descriptor is valid
+        let file = unsafe { File::from_raw_fd(fd) };
+        let mmap_size = (last_ram_addr / (VHOST_LOG_PAGE * 8)) + 1;
+        let mmap_handle = file.as_raw_fd();
+
+        // Set shm_log region size
+        file.set_len(mmap_size).map_err(Error::SetFileSize)?;
+
+        // Set the seals
+        let res = unsafe {
+            libc::fcntl(
+                file.as_raw_fd(),
+                libc::F_ADD_SEALS,
+                libc::F_SEAL_GROW | libc::F_SEAL_SHRINK | libc::F_SEAL_SEAL,
+            )
+        };
+        if res < 0 {
+            return Err(Error::Fcntl(std::io::Error::last_os_error()));
```

Comment:
I think this would be a better as `Error::SetSeals`

Reply:
Agreed!